tls/ttls.c

```diff
@@ -1760,7 +1766,7 @@ ttls_write_finished(TlsCtx *tls, struct sg_table *sgt, unsigned char **in_buf)
 
 	sg_init_table(&sg, 1);
 	sg_set_buf(&sg, p, TLS_HEADER_SIZE + TTLS_HS_FINISHED_BODY_LEN);
-	if ((r = ttls_encrypt(tls, &enc_sgt)))
+	if ((r = ttls_encrypt(tls, &enc_sgt, NULL)))
```

Comment:
If you use `&enc_sgt` for both second and third parameter here, `if (out_sgt)` checks from `ttls_encrypt()` could be removed.